src/images.cpp

```diff
@@ -1,31 +1,116 @@
+      // printf("DEBUG : jd(%d = floor(%.8f)) = %.8f\n",(int)uxtime,uxtime_d,jd_out);
+      jd_out += (uxtime_d - uxtime)/(24.00*3600.00);
+      // printf("DEBUG : jd_prim = %.8f\n",jd_out);
+
+      // which to use ???
+      double sid_greenwich_h = ln_get_apparent_sidereal_time(jd_out);
+      // printf("DEBUG : sid_greenwich = %.8f [h] = %.8f [deg]\n",sid_greenwich_h,sid_greenwich_h*15.00);
+  
+      double sid_local_h = sid_greenwich_h + geo_long_deg/15.00;
+      // printf("DEBUG : sid_local = %.8f + %.8f = %.8f [h] = %.8f [deg]\n",sid_greenwich_h,geo_long_deg/15.00,sid_local_h,sid_local_h*15.00);
+
+      cut_to_range(sid_local_h);
+
+      return sid_local_h;  
+   }
+
+   void fixCoordHdr( double ra_center_deg, double dec_center_deg, double lst_hours, double long_deg, double lat_deg, double& xi, double& eta )
+   {
+      const double deg2rad = (M_PI/180.00);
+
+      double lat_radian = lat_deg*deg2rad;
+      double long_radian = long_deg*deg2rad;
+
+      double dec_center_rad=dec_center_deg*deg2rad;
+      double ra_center_rad=ra_center_deg*deg2rad;
+      double ra_center_h=ra_center_deg/15.0;
+
+      double ha_hours=lst_hours-ra_center_h;
+      double ha_radians=(ha_hours*15.00)*deg2rad;
+
+      // printf("values = %.8f %.8f %.8f , %.8f\n",lat_radian,ra_center_rad,dec_center_rad,ha_radians);
+      double cosZ = sin(lat_radian)*sin(dec_center_rad) + cos(lat_radian)*cos(dec_center_rad)*cos(ha_radians);
+      double tanZ = sqrt(1.00-cosZ*cosZ)/cosZ;
+
+      // printf("tanZ = %.8f\n",tanZ);
+
+      // Parallactic angle
+      // http://www.gb.nrao.edu/~rcreager/GBTMetrology/140ft/l0058/gbtmemo52/memo52.html
+      double tan_chi = sin(ha_radians)/( cos(dec_center_rad)*tan(lat_radian) - sin(dec_center_rad)*sin(ha_radians)  );
+
+      // $lat_radian $dec_radian $ha_radian
+      // printf("DEBUG : values2 : %.8f %.8f %.8f\n",lat_radian,dec_center_rad,ha_radians);
+      double chi_radian = atan2( sin(ha_radians) , cos(dec_center_rad)*tan(lat_radian) - sin(dec_center_rad)*cos(ha_radians) );
+
+      // printf("chi_radian = %.8f\n",chi_radian);
+
+      // there is a - sign in the paper, but Randall says it's possibly wrong:
+      // so I stay with NO - SIGN version
+      xi=tanZ*sin(chi_radian);
+      eta=tanZ*cos(chi_radian);
+
+      // printf("xi = %.8f\n",xi);
+      // printf("eta = %.8f\n",eta);
+   }    
+}
+
+void Images::save_fits_file(const std::string filename, float* data, long side_x, long side_y){
+    FITS fitsImage;
+    FITS::HDU hdu;
+    hdu.set_image(data,  side_x, side_y);
     
-        fitsImage.add_HDU(hdu);
-        fitsImage.to_file(filename);
-    }
+    // calculate LST :
+    double jd,xi,eta;
+    double lst_hours = get_local_sidereal_time( obsInfo.startTime, obsInfo.geo_long_deg, jd );
+    fixCoordHdr( ra_deg, dec_deg, lst_hours, obsInfo.geo_long_deg, obsInfo.geo_lat_deg, xi, eta );
+    
+    
+    // hdu.add_keyword("TIME", static_cast<long>(obsInfo.startTime), "Unix time (seconds)");
```

Comment:
Could you remove these commented out statements? I had kept them just for reference.. they are not needed.